src/test/ui/resolve/token-error-correct-3.rs

```diff
@@ -21,15 +21,12 @@ pub mod raw {
                                                                callback: F)
                                                                -> io::Result<bool> {
         if !is_directory(path.as_ref()) { //~ ERROR: cannot find function `is_directory`
-            callback(path.as_ref(); //~ ERROR expected one of
-            fs::create_dir_all(path.as_ref()).map(|()| true) //~ ERROR: mismatched types
-            //~^ expected (), found enum `std::result::Result`
-            //~| expected type `()`
-            //~| found type `std::result::Result<bool, std::io::Error>`
-            //~| expected one of
-        } else { //~ ERROR: incorrect close delimiter: `}`
-            //~^ ERROR: expected one of
-            //~| unexpected token
+            callback(path.as_ref();
+            //~^ ERROR expected one of
+            //~| ERROR expected expression
+            fs::create_dir_all(path.as_ref()).map(|()| true)
+        } else {
+            //~^ ERROR incorrect close delimiter: `}`
             Ok(false);
         }
```

Comment:
What would be the *ideal* output for this file?

I'll get back to this over the weekend to see if I can make the parser recover from missing delimiters and parse the rest correctly, so that we get the type error about `fs::create_dir_all` back.